lib/uniris/oracles/scheduler.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Uniris.Oracles.Scheduler do
+  @moduledoc false
+
+  use GenServer
+
+  alias Uniris.Oracles.TransactionContent
+
+  alias Uniris.TransactionChain.{
+    Transaction,
+    TransactionData
+  }
+
+  alias Uniris.Utils
+
+  # Public
+
+  @spec start_link(mfa: mfa(), interval: String.t()) ::
+          {:ok, pid()} | :ignore | {:error, {:already_started, pid()} | term()}
+  def start_link(mfa: mfa, interval: interval) do
+    ms_interval = parse_interval(interval)
+
+    GenServer.start_link(__MODULE__, mfa: mfa, interval: ms_interval)
+  end
+
+  @spec get_payload(pid()) :: {:ok, map()}
+  def get_payload(pid), do: GenServer.call(pid, :get_payload)
+
+  # Implementation
+
+  @impl GenServer
+  def init(mfa: mfa, interval: interval) do
+    state = %{
+      mfa: mfa,
+      interval: interval,
+      subscribers: [],
+      payload: nil
+    }
+
+    {:ok, state, {:continue, :prepare}}
+  end
+
+  @impl GenServer
+  def handle_continue(:prepare, state = %{mfa: {m, _, _}}) do
+    m.start()
+
+    Process.send_after(self(), :fetch, 5_000)
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_call(:get_payload, subscriber, state = %{payload: nil, subscribers: subscribers}) do
```

Comment:
I think we can leverage `Registry` to support subscribers to avoid timeouts and GenServer bottlenecks using a asynchronous communication mechanism, like in `Uniris.PubSub`.

So contracts will register directly to a topic for example `{:oracle, :uco_price}` and will receive the good message after the payload retrieval.